flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java

```diff
@@ -0,0 +1,105 @@
+import static org.apache.flink.formats.csv.CsvFormatOptions.DISABLE_QUOTE_CHARACTER;
+import static org.apache.flink.formats.csv.CsvFormatOptions.ESCAPE_CHARACTER;
+import static org.apache.flink.formats.csv.CsvFormatOptions.FIELD_DELIMITER;
+import static org.apache.flink.formats.csv.CsvFormatOptions.IGNORE_PARSE_ERRORS;
+import static org.apache.flink.formats.csv.CsvFormatOptions.NULL_LITERAL;
+import static org.apache.flink.formats.csv.CsvFormatOptions.QUOTE_CHARACTER;
+
+/** A class with common CSV format constants and utility methods. */
+class CsvCommons {
+
+    public static final String IDENTIFIER = "csv";
+
+    // ------------------------------------------------------------------------
+    //  Validation
+    // ------------------------------------------------------------------------
+
+    static void validateFormatOptions(ReadableConfig tableOptions) {
+        final boolean hasQuoteCharacter = tableOptions.getOptional(QUOTE_CHARACTER).isPresent();
+        final boolean isDisabledQuoteCharacter = tableOptions.get(DISABLE_QUOTE_CHARACTER);
+        if (isDisabledQuoteCharacter && hasQuoteCharacter) {
+            throw new ValidationException(
+                    "Format cannot define a quote character and disabled quote character at the same time.");
+        }
+        // Validate the option value must be a single char.
+        validateCharacterVal(tableOptions, FIELD_DELIMITER, true);
+        validateCharacterVal(tableOptions, ARRAY_ELEMENT_DELIMITER);
+        validateCharacterVal(tableOptions, QUOTE_CHARACTER);
+        validateCharacterVal(tableOptions, ESCAPE_CHARACTER);
+    }
+
+    /** Validates the option {@code option} value must be a Character. */
+    private static void validateCharacterVal(
+            ReadableConfig tableOptions, ConfigOption<String> option) {
+        validateCharacterVal(tableOptions, option, false);
+    }
+
+    /**
+     * Validates the option {@code option} value must be a Character.
+     *
+     * @param tableOptions the table options
+     * @param option the config option
+     * @param unescape whether to unescape the option value
+     */
+    private static void validateCharacterVal(
+            ReadableConfig tableOptions, ConfigOption<String> option, boolean unescape) {
+        if (!tableOptions.getOptional(option).isPresent()) {
+            return;
+        }
+
+        final String value =
+                unescape
+                        ? StringEscapeUtils.unescapeJava(tableOptions.get(option))
+                        : tableOptions.get(option);
+        if (value.length() != 1) {
+            throw new ValidationException(
+                    String.format(
+                            "Option '%s.%s' must be a string with single character, but was: %s",
+                            IDENTIFIER, option.key(), tableOptions.get(option)));
+        }
+    }
+
+    public static Set<ConfigOption<?>> optionalOptions() {
```

Comment:
Can you align the options with this PR https://github.com/apache/flink/pull/18290 ?